Hudl.Mjolnir/Config/MjolnirConfiguration.cs

```diff
@@ -0,0 +1,128 @@
+        /// Default constructor just to create dictionaires.
+        /// </summary>
+        protected MjolnirConfiguration()
+        {
+            CommandConfigurations = new Dictionary<string, CommandConfiguration>();
+            DefaultCommandConfiguration = new CommandConfiguration();
+            BulkheadConfigurations = new Dictionary<string, BulkheadConfiguration>();
+            DefaultBulkheadConfiguration = new BulkheadConfiguration();
+            BreakerConfigurations = new Dictionary<string, BreakerConfiguration>();
+            DefaultBreakerConfiguration = new BreakerConfiguration();
+        }
+        
+        /// <summary>
+        /// Gets command configuration for a given key. If key is null or not exists in configuration dictionary
+        /// default configuration is being returned.
+        /// </summary>
+        /// <param name="groupKey">Command configuration for a given key. Default value returned if non-existent or 
+        /// null.</param>
+        /// <returns></returns>
+        public CommandConfiguration GetCommandConfiguration(string groupKey = null)
+        {
+            if (groupKey == null) return DefaultCommandConfiguration;
+            CommandConfiguration commandConfiguration;
+            return CommandConfigurations.TryGetValue(groupKey, out commandConfiguration) ?
+                commandConfiguration : DefaultCommandConfiguration;
+        }
+        
+        /// <summary>
+        /// Gets bulkhead configuration for a given key. If key is null or not exists in configuration dictionary 
+        /// default configuration is being returned.
+        /// </summary>
+        /// <param name="key">Bulkhead configuration for a given key. Default value returned if non-existent or 
+        /// null.</param>
+        /// <returns></returns>
+        public BulkheadConfiguration GetBulkheadConfiguration(string key = null)
+        {
+            if (key == null) return DefaultBulkheadConfiguration;
+            BulkheadConfiguration bulkheadConfiguration;
+            return BulkheadConfigurations.TryGetValue(key, out bulkheadConfiguration) ?
+                bulkheadConfiguration : DefaultBulkheadConfiguration;
+        }       
+        
+        
+        /// <summary>
+        /// Gets breaker configuration for a given key. If key is null or not exists in configuration dictionary 
+        /// default configuration is being returned.
+        /// </summary>
+        /// <param name="key">Breaker configuration for a given key. Default value returned if non-existent or 
+        /// null.</param>
+        /// <returns></returns>
+        public BreakerConfiguration GetBreakerConfiguration(string key = null)
+        {
+            if (key == null) return DefaultBreakerConfiguration;
+            BreakerConfiguration breakerConfiguration;
+            return BreakerConfigurations.TryGetValue(key, out breakerConfiguration) ?
+                breakerConfiguration : DefaultBreakerConfiguration;
```

Comment:
If someone adds a new `BreakerConfiguration` to the `Dictionary` after the initialization (e.g. the config didn't exist, but then someone went and added it via dynamic config), what's the behavior? Are there thread safety concerns if the dictionaries are modified? Should the dictionaries be the `Concurrent` versions? Same applies to other config dictionaries.